diff -Naur darktable-0.6.orig/darktable.desktop darktable-0.6/darktable.desktop
--- darktable-0.6.orig/darktable.desktop	2010-08-27 18:46:31.000000000 +0200
+++ darktable-0.6/darktable.desktop	2010-08-31 08:23:59.031379181 +0200
@@ -15,7 +15,7 @@
 Comment[nl]=Organiseer en ontwikkel beelden van digitale camera's
 Comment[fi]=Järjestele ja kehitä kuvia digitaalikameroista
 Comment[fr]=Organiser et développer les images d'un boîtier numérique
- 
+
 Version=1.0
 
 Type=Application
