diff -Naur darktable-1.0~rc2.ori/src/common/curve_tools.c darktable-1.0~rc2/src/common/curve_tools.c
--- darktable-1.0~rc2.ori/src/common/curve_tools.c	2012-03-10 00:44:17.000000000 +0100
+++ darktable-1.0~rc2/src/common/curve_tools.c	2012-03-10 13:12:24.000000000 +0100
@@ -753,8 +753,8 @@
 {
   int i = 0, n;
 
-  float x[20];
-  float y[20];
+  float x[20] = {0};
+  float y[20] = {0};
   float *ypp;
 
   //The box points  are what the anchor points are relative
